trie.c

```diff
@@ -45,7 +45,7 @@ bool st_trie_get_completion(st_cursor_t *cursor, st_trie_search_result_t *res)
     st_cursor_init(cursor, 0, false);
     st_find_longest_chain(cursor, &res->trie_match, 0);
 #if SEQUENCE_TRANSFORM_FALLBACK_BUFFER
-    if (st_cursor_init(cursor, 0, true)) {
+    if (!res->trie_match.is_chained_match && st_cursor_init(cursor, 0, true)) {
```

Comment:
If a chained match was found in the initial search, skip the fallback search on the vout cursor.